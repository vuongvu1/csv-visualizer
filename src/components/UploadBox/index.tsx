import React from "react";
import SC from "./styles";

interface Props {}

const UploadBox: React.FC<Props> = () => {
  return <SC.Container>UPload here</SC.Container>;
};

export default UploadBox;
