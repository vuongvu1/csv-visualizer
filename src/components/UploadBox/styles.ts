import styled from "styled-components";
import { Spacing, ScreenWidth } from "configs/styles";
import {
  whiteColor,
  boxShadowColor,
  secondaryColor,
  neutralDarkColor,
  secondaryLightColor,
} from "configs/colors";
import { Document as DocumentIcon } from "assets/icons";

export default {
  Container: styled.div`
    flex: 1;
    display: flex;
    flex-direction: column;
    margin: ${Spacing.large};
    margin-right: 0;
    background-color: ${whiteColor};
    box-shadow: 0px 1px 0px ${boxShadowColor};
    border-radius: 12px;

    @media (max-width: ${ScreenWidth.medium}) {
      margin-right: ${Spacing.large};
      margin-bottom: 0;
    }
  `,
  FileInput: styled.input`
    display: none;
  `,
  Label: styled.label`
    flex: 1;
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    margin: ${Spacing.normal};
    border-radius: 12px;
    border: 1px dashed ${secondaryColor};
    color: ${neutralDarkColor};
    cursor: pointer;
    transition: background-color 0.3s;
    text-align: center;

    &:hover {
      background-color: ${secondaryLightColor};
    }
  `,
  UploadedFile: styled.div`
    margin: ${Spacing.normal} ${Spacing.normal} 0;
    display: flex;
    justify-content: space-between;
  `,
  IconAndLabel: styled.div`
    display: flex;
    align-items: center;
  `,
  FileIcon: styled(DocumentIcon)`
    margin-right: ${Spacing.small};
  `,
};
