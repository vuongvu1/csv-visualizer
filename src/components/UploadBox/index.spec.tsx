import React from "react";
import { render, cleanup } from "@testing-library/react";
import UploadBox from "./index";

describe("UploadBox", () => {
  afterEach(cleanup);

  test("should match snapshot", () => {
    const { container } = render(<UploadBox />);

    expect(container).toMatchSnapshot();
  });
});
