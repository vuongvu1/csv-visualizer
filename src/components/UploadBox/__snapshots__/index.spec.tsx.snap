// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UploadBox should match snapshot 1`] = `
.c0 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin: 18px;
  margin-right: 0;
  padding: 12px;
  background-color: #FFFFFF;
  box-shadow: 0px 1px 0px rgba(165,165,168,0.2);
  border-radius: 12px;
}

.c2 {
  display: none;
}

.c1 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border-radius: 12px;
  border: 1px dashed #63AEF7;
  color: #707070;
  cursor: pointer;
  -webkit-transition: background-color 0.3s;
  transition: background-color 0.3s;
  text-align: center;
  padding: 12px;
}

.c1:hover {
  background-color: #d9eafb;
}

@media (max-width:768px) {
  .c0 {
    margin-right: 18px;
    margin-bottom: 0;
  }
}

<div>
  <div
    class="c0"
  >
    <label
      class="c1"
      for="file-id"
    >
      Click to upload
      <br />
      (accept only .csv file)
    </label>
    <input
      accept=".csv"
      class="c2"
      id="file-id"
      type="file"
    />
  </div>
</div>
`;
